soda-consumer-scala/src/main/scala/com/socrata/soda2/consumer/impl/LegacyRowDecoder.scala

```diff
@@ -27,12 +26,15 @@ private[consumer] class LegacyRowDecoder(datasetBase: URI, rawSchema: Map[Column
 }
 
 private[consumer] object LegacyRowDecoder {
-  val conversionForType: ((ColumnName, String)) => (ColumnName, SodaType) = { case (col, rawType) =>
-    col -> typeMap.getOrElse(rawType, throw new UnknownTypeException(rawType))(col.toString)
+  val conversionForType: PartialFunction[(ColumnName, String), (ColumnName, SodaType)] = {
+    case (col, rawType) if nonDeprecated(rawType) =>
+      col -> typeMap.getOrElse(rawType, throw new UnknownTypeException(rawType))(col.toString)
```

Comment:
I'd generally prefer to make this a total function `((ColumnName, String)) => Option[(ColumnName, SodaType)]` and use `flatMap` instead of `collect` where applicable.